File: app/src/main/java/gb/android/notes2/model/NoteListItemSource.java
package gb.android.notes2.model;

public interface NoteListItemSource {
    void updateData();

    int size();

    NoteListItem getNoteListItemByPos(int position);

    NoteListItem getNoteListItemById(int id);

    void setSort(int sortType);

    void addNote();

    void deleteNote(int position);

    void deleteAll();

    String getNoteTextById(int id);

    void updateNoteItemById(int id, String title, String date);

    void updateNoteTextById(int id, String text);
}

File: app/src/main/java/gb/android/notes2/model/NoteListItemSource.kt
package gb.android.notes2.model;

public interface NoteListItemSource {
    NoteListItemSource init (NoteListSourceResponse noteListSourceResponse);

    void updateData();

    int size();

    NoteListItem getNoteListItemByPos(int position);

    void requestNoteListItemById(String id);

    void setSort(int sortType);

    void addNote();

    void deleteNote(String position);

    void deleteAll();

    void updateNoteItemById(String id, String title, String date);

    void updateNoteTextById(String id, String text);
}
